src/main/java/seedu/address/model/VersionedAddressBook.java

```diff
@@ -65,16 +102,20 @@ public boolean equals(Object o) {
         if (o == null) {
             return false;
         }
+        if (!(o instanceof VersionedAddressBook)) {
+            return false;
+        }
         VersionedAddressBook other = (VersionedAddressBook) o;
 
         // Compare addressBookStateList and currentStatePointer
         return currentStatePointer == other.currentStatePointer
-                && addressBookStateList.equals(other.addressBookStateList);
+                && addressBookStateList.equals(other.addressBookStateList)
+                && current.equals(other.current);
```

Comment:
I am not sure about this part actually, do we need to compare uncommitted changes (current) as well?